test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -194,7 +194,8 @@ spec =
             list.add 123
             list_tpe = Meta.type_of list
             list_tpe . should_not_equal_type JObject
-            list_tpe . should_equal_type ArrayList
+            list_tpe . should_not_equal_type ArrayList
+            list_tpe . should_equal_type Array
```

Comment:
For purposes of [ascribed type runtime checks](https://github.com/enso-org/enso/pull/6790) it is better if polyglot values that look like Enso values get Enso type - e.g. everything that `hasArrayElements` should have type `Array`.

Reply:
That makes the experience indeed much more consistent!

On the other hand, in some rarer cases it may be good to be able to learn the 'true underlying Truffle type' - so as we discussed in today's standup, it may be worth to, at some point, add a `Polyglot.get_meta_object` method that would allow to access this.